### PR TITLE
Deploy: communicate closure size

### PR DESCRIPTION
**Pull request by domenkozar**
I wasn't able to make `shellOut` to return `stdout` so I create a similar function with a slightly different behaviour :sweat_smile: 

While I was at it, I refactored all cache setup into a separate function.

### CODE REVIEW

File: cachix/src/Cachix/Deploy/Activate.hs [hunk 1]
```diff
@@ -112,16 +88,21 @@ activate options connection sourceStream deploymentDetails agentInfo agentToken
           if not (all (== ExitSuccess) exitCodes)
             then deploymentFailed
             else do
-              now <- liftIO getCurrentTime
-              sendMessage $ deploymentFinished True now
+              now2 <- liftIO getCurrentTime
```
Comment (sandydoo) [inline cachix/src/Cachix/Deploy/Activate.hs:91]: 🙈

File: cachix/src/Cachix/Deploy/Activate.hs [hunk 2]
```diff
@@ -8,11 +9,18 @@ import qualified Cachix.Client.NetRc as NetRc
 import qualified Cachix.Deploy.Websocket as CachixWebsocket
 import qualified Cachix.Types.BinaryCache as BinaryCache
 import Cachix.Types.Permission (Permission (..))
+import qualified Data.Aeson as Aeson
+#if MIN_VERSION_aeson(2,0,0)
```
Comment (sandydoo) [inline cachix/src/Cachix/Deploy/Activate.hs:13]: Do we have to support other versions of aeson?

Comment (domenkozar) [reply cachix/src/Cachix/Deploy/Activate.hs:13]: The backend is still running aeson-1

Comment (sandydoo) [reply cachix/src/Cachix/Deploy/Activate.hs:13]: Ah, I see! It consumes this as a library.

**Pull request merged**